ShoppingList/src/main/java/edu/agh/eaiib/model/ProductList.java

```diff
@@ -2,26 +2,26 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
+
+public class ProductList extends ArrayList<Product> {
```

Comment:
Why extends? I usually follow "Favor composition over inheritance" rule.

Reply:
Another thing: please run repository tests and check if it still works 😺

Reply:
It is easier to use this way. And also it is true. I mean, for me it is just a list with name.

Reply:
But:
(essay)
1) We get from ArrayList a lot of methods that are now part of public API of our class - are we sure we want them all? E.g. there's a lot of methods to modify data instead of single one to do it so it's harder to find how object changed while debugging. Also some methods give incorrect results e.g. when calling `toArray()` we lost data.
2) We are tied to this ArrayList and can't change implementation easily - e.g we can want to change it to HashMap/List<List> or sth internally.
And it's not only list with name - it's complicated structure now! 😄